synapse/lib/ast.py

```diff
@@ -3083,6 +3096,9 @@ class Bool(Const):
 
 class EmbedQuery(Const):
 
+    def isSafeEdit(self):
```

Comment:
is `isSafeEdit` mostly synonymous with anything more generic? Technically could a subquery assignment with no edit blocks be isSafeEdit() = True?

Reply:
Yeah there are subqueries which could be safe with a bit of runtime validation but was keeping it simpler while working on it.

Reply:
Might be a better more generic name for it but I'm still working through the best things to check for "safe" edits because various subqueries can potentially be safe depending on runtime state